ecs-agent/manageddaemon/managed_daemon.go

```diff
@@ -0,0 +1,326 @@
+// of valid ManagedDeamon objects
+func ImportAll() []*ManagedDaemon {
+	// TODO parse taskdef json files in /deps/daemons
+	// TODO validate that each daemon has a corresponding image tar
+	ebsManagedDaemon := NewManagedDaemon("ebs-csi-driver",
+		"public.ecr.aws/ebs-csi-driver/aws-ebs-csi-driver",
+		"v1.20.0")
+	// TODO add healthcheck
+	// TODO add mount points
+	return []*ManagedDaemon{ebsManagedDaemon}
+}
+
+func (md *ManagedDaemon) SetHealthCheck(
+	healthCheckTest []string,
+	healthCheckInterval time.Duration,
+	healthCheckTimeout time.Duration,
+	healthCheckRetries int) {
+	md.healthCheckInterval = healthCheckInterval
+	md.healthCheckTimeout = healthCheckTimeout
+	md.healthCheckRetries = healthCheckRetries
+	md.healthCheckTest = make([]string, len(healthCheckTest))
+	copy(md.healthCheckTest, healthCheckTest)
+}
+
+func (md *ManagedDaemon) GetImageName() string {
+	return md.imageName
+}
+
+func (md *ManagedDaemon) GetImageRef() string {
+	return md.imageRef
+}
+
+func (md *ManagedDaemon) GetImageTag() string {
+	return md.imageTag
+}
+
+func (md *ManagedDaemon) GetImageCanonicalRef() string {
+	return (fmt.Sprintf("%s:%s", md.imageRef, md.imageTag))
+}
+
+func (md *ManagedDaemon) GetImageTarPath() string {
+	return (fmt.Sprintf("%s/%s", imageTarPath, md.imageName))
+}
+
+func (md *ManagedDaemon) GetAgentCommunicationMount() *MountPoint {
+	return md.agentCommunicationMount
+}
+
+func (md *ManagedDaemon) GetApplicationLogMount() *MountPoint {
+	return md.applicationLogMount
+}
+
+// returns list of mountpoints without the
+// agentCommunicationMount and applicationLogMount
+func (md *ManagedDaemon) GetFilteredMountPoints() []*MountPoint {
+	filteredMounts := make([]*MountPoint, len(md.mountPoints))
+	copy(filteredMounts, md.mountPoints)
+	return filteredMounts
+}
+
+// returns list of mountpoints which (re)integrates
+// agentCommunicationMount and applicationLogMount
+// these will always include host mount file overrides
+func (md *ManagedDaemon) GetMountPoints() []*MountPoint {
+	allMounts := make([]*MountPoint, len(md.mountPoints))
+	copy(allMounts, md.mountPoints)
+	allMounts = append(allMounts, md.agentCommunicationMount)
+	allMounts = append(allMounts, md.applicationLogMount)
+	return allMounts
+}
+
+func (md *ManagedDaemon) GetEnvironment() map[string]string {
+	return md.environment
+}
+
+func (md *ManagedDaemon) GetLoadedDaemonImageRef() string {
+	return md.loadedDaemonImageRef
+}
+
+// filter mount points for agentCommunicationMount
+// set required mounts
+// and override host paths in favor of agent defaults
+// when a duplicate SourceVolumeID is given, the last Mount wins
+func (md *ManagedDaemon) SetMountPoints(mountPoints []*MountPoint) error {
+	var mountPointMap = make(map[string]*MountPoint)
+	for _, mp := range mountPoints {
+		if mp.SourceVolumeID == defaultAgentCommunicationMount {
+			mp.SourceVolumeHostPath = fmt.Sprintf("%s/%s/", defaultAgentCommunicationPathHostRoot, md.imageName)
```

Comment:
No `ContainerPath`? Is this handled by others?

Reply:
https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/managed_daemon/mountpoint.go#L26 <- this will be part of the mountpoint.  We are just overriding the host mount point to force it to where the agent expects it to be.